apps/migrations/0002_initial.py

```diff
@@ -63,6 +63,11 @@ class Migration(migrations.Migration):
             name='project',
             field=models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, related_name='appinstance', to='projects.project'),
         ),
+        migrations.AddField(
+            model_name='appinstance',
+            name='flavor',
+            field=models.ForeignKey(on_delete=django.db.models.deletion.RESTRICT, related_name='appinstance', to='projects.flavor'),
+        ),
```

Comment:
There is a hierarchy:
Appinstance ---has---> Flavor ---has---> Project.
I've added a restriction so that deleting a flavor that is used by an appinstance is not permitted. However, deleting a project, is allowed because it deletes both the flavor and appinstance.

Reply:
Yes, the PR looks good now. I haven't had time to test it manually though.